src/topmodel.c

```diff
@@ -508,9 +508,189 @@ if(yes_print_output==TRUE)
 return;
 }
 
+
+/*********************************
+ * Define functions to be used in init() and in Set_value in bmi_topmodel.c
+ * to enable calibratable parameters to be updated
+ * **********************************/
+
+
+// FUNCTION TO CONVERT DISTANCE/AREA FORM TO TIME DELAY HISTOGRAM ORDINATES
+// outputs used in subsequent functions include tch
+extern void convert_dist_to_histords(double *dist_from_outlet, int num_channels,
+					double chv, double rv, double dt, double tch[11])
+{
+    // declare local variables
+    double chvdt, rvdt;
+    int j;
+    
+    /*  Convert parameters to m/time step DT
+    *  with exception of XK0 which must stay in m/h
+    *                    Q0 is already in m/time step
+                     T0 is input as Ln(To) */
+
+    chvdt = chv*dt;
+    rvdt = rv*dt;
+
+    tch[1]=dist_from_outlet[1]/chvdt;
+    for(j=2;j<=num_channels;j++)
+     {
+      tch[j]=tch[1]+(dist_from_outlet[j]-dist_from_outlet[1])/rvdt;
+     }
+
+    return;
+
+}
+
+// FUNCTION TO CALCULATE THE TIME_DELAY_HISTOGRAM
+// updates values associated with model structure
+// outputs used in TOPMOD() include num_time_delay_histo_ords, time_delay_histogram,
+// num_delay, and Q
+extern void calc_time_delay_histogram(int max_time_delay_ordinates, int num_channels, double area,
+				double tch[11], int *num_time_delay_histo_ords, int *num_delay,
+				double **time_delay_histogram, double *cum_dist_area_with_dist) 
+
+
+{
+
+    // declare local variables
+    double time, a1, sumar, a2;
+    int j, ir;
+
+    if((*time_delay_histogram)==NULL)
+      {
+      d_alloc(time_delay_histogram,max_time_delay_ordinates);
+      }
+
+
+    (*num_time_delay_histo_ords)=(int)tch[num_channels];
+    
+    
+    if((double)(*num_time_delay_histo_ords)<tch[num_channels]) 
```

Comment:
This condition is impossible to hit in a serial execution since you just assigned this value to be `tch[num_channels]` in the previous line...